pyathena/connection.py

```diff
@@ -23,7 +23,11 @@
 _logger = logging.getLogger(__name__)  # type: ignore
 
 
-class Connection:
+ConnectionCursor = TypeVar("ConnectionCursor", bound=BaseCursor)
+FunctionalCursor = TypeVar("FunctionalCursor", bound=BaseCursor)
```

Comment:
What is the reason for separating the two types, ConnectionCursor and FunctionalCursor? It seems to me that it would be better to use one of the definitions. (Perhaps just defining a ConnectionCursor would be enough.)

Reply:
The return type of the `def cursor` method depends on two types: the `cursor_class` parameter in the constructor of Connection and the `cursor` parameter itself. I attempted to guide the type analyzer in classifying two behaviors:
- When the cursor parameter is omitted (`def cursor()`), the return type is `ConnectionCursor`.
- When the cursor parameter is specified (`def cursor(MyCursor)`), the return type is expected to be `MyCursor`.

However, regrettably, the type analyzer faces confusion in the following case:
- When the cursor parameter is specified (`def cursor(MyCursor)`), the inferred return type becomes `MyCursor | ConnectionCursor`.

I now believe that it's better to stick with only `FunctionalCursor` and eliminate the generic `ConnectionCursor`.

Reply:
Understood. I think it is a good implementation. Let's keep the current implementation.